### PR TITLE
Move to simpler task patterns.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/76446

### CODE REVIEW

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingIdleProcessor.cs [hunk 1]
```diff
@@ -117,6 +118,9 @@ protected async Task<bool> WaitForIdleAsync(IExpeditableDelaySource expeditableD
 
     private async Task ProcessAsync()
     {
+        // Ensurew we do this processing on the background.
+        await Task.Yield().ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingIdleProcessor.cs:122]: `await TaskScheduler.Default;` not available here as we're at teh Features layer and don't have access to that VS extension.

File: src/Workspaces/Core/Portable/Shared/TestHooks/TaskExtensions.cs [hunk 2]
```diff
@@ -36,13 +36,16 @@ public static Task CompletesTrackingOperation(this Task task, IDisposable token)
...
 
-        static Task CompletesTrackingOperationSlow(Task task, IDisposable token)
+        static async Task CompletesTrackingOperationSlow(Task task, IDisposable token)
         {
-            return task.SafeContinueWith(
-                t => token.Dispose(),
-                CancellationToken.None,
-                TaskContinuationOptions.ExecuteSynchronously,
-                TaskScheduler.Default);
+            try
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/Shared/TestHooks/TaskExtensions.cs:41]: Any reason to not do this?

```
using var _ = token;

await task.NoThrowAwaitableInternal(captureContext: false);
```

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Shared/TestHooks/TaskExtensions.cs:41]: wfm.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide ptal.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide @jasonmalinowski ptal

**Review by JoeRobich — APPROVED** [review]

**Pull request merged**